src/optimization/objective.py

```
@@ -0,0 +1,73 @@
+    config_data['training']['max_epochs'] = config_data['tuning']['epochs_per_trial']
+    config_data['training']['patience'] = config_data['tuning']['patience_per_trial']
+
+    # --- 3. Validate the dynamic config and run the pipeline ---
+    try:
+        # Ensure model constraints are met (e.g., hidden_size must be divisible by num_heads)
+        if hidden_size % num_heads != 0:
+            raise optuna.TrialPruned(f"hidden_size {hidden_size} not divisible by num_attention_heads {num_heads}.")
+        
+        config = AppConfig(**config_data)
+    except Exception as e:
+        raise optuna.TrialPruned(f"Configuration validation failed: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)